toolkit/tools/internal/packagerepo/repocloner/rpmrepocloner/rpmrepocloner.go

```diff
@@ -24,10 +25,13 @@ import (
 
 const (
 	allRepoIDs             = "*"
+	marinerReposGlob       = "mariner*"
```

Comment:
This method disables any repository passed using REPO_LIST but starting "mariner". For example if I use this flag and pass `mariner-extended`, it will be disabled.

Reply:
Just had a thought... could we call `rpm -e mariner-repos-*` or something similar? Just fully remove the repo files that way... The user call always explicitly re-add any repo file they wish that way.

Reply:
This might have unexpected consequences due to removal of `mariner-repos-shared` (package holds GPG keys, which may be needed). But I think @0xba1a's main concern is to make sure we remove only the things we need. My one idea would be to modify our logic, where for `DISABLE_MARINER_REPOS=y` we first remove all .repo files except for the ones built into the chroot (the ones representing the local caches) and then proceed to add the ones specified by the user.

What do you think?

Reply:
Update: I think we may both simplify and make sure we include the right repos by only modifying the `initializeRepoDefinitions` function. More details in my comment around `if r.disableMarinerRepos`.

Reply:
Instead of removing the repos based on re, can we introduce a flag (`DISABLE_REPO_LIST`) using which the user can pass repo-ids she wants to disable explicitly?